### ISSUE TITLE
Add support for Ollama

### ISSUE THREAD

**Issue by KTheMan** [TheComamba/UnKenny#76]
I have Ollama instance set up on an external server and think it would be a great, easy fit for this type of add-on if I could hook it in. Opens up use of many more models with better hardware, if the GM has access, IMO

**Comment by KingphilAusti** [comment]
So maybe a clearly defined interace to an external model could solve this. 

I don't know if we can bulild this as an "easy to use for everyone" solution but could provide an interface which connects to some external model via a few lines of code.

**Comment by KTheMan** [comment]
Yup, pretty much what I was thinking; shouldn't need much, as Ollama exposes an OpenAI compatible endpoint.

From [https://ollama.com/blog/openai-compatibility](https://ollama.com/blog/openai-compatibility):
```javascript
import OpenAI from 'openai'

const openai = new OpenAI({
  baseURL: 'http://localhost:11434/v1/',

  // required but ignored
  apiKey: 'ollama',
})

const chatCompletion = await openai.chat.completions.create({
  messages: [{ role: 'user', content: 'Say this is a test' }],
  model: 'llama3',
})
```

**Comment by KingphilAusti** [comment]
Ah I see. Going for a fork right now was also what I would have suggested. I should be enough to change openai-api, as you did. 
What is the error you get? Or how does your response look like?

**Comment by KingphilAusti** [comment]
Ah I see. Going for a fork right now was also what I would have suggested. I should be enough to change openai-api, as you did. 
What is the error you get? Or how does your response look like?
